packages/sfui/frameworks/vue/components/SfCheckbox/SfCheckbox.vue

```diff
@@ -1,39 +1,66 @@
 <script setup lang="ts">
-import type { InputHTMLAttributes, PropType } from 'vue';
-import { computed, toRefs } from 'vue';
+import { nextTick, ref, useTemplateRef, watch, type InputHTMLAttributes } from 'vue';
+import { ClassProp, SfIconCheckBox, SfIconCheckBoxOutlineBlank, SfIconIndeterminateCheckBox } from '@storefront-ui/vue';
 
-const props = defineProps({
-  modelValue: {
-    type: [String, Array, Boolean] as PropType<InputHTMLAttributes['checked']>,
+const { indeterminate = false } = defineProps({
+  invalid: {
+    type: Boolean,
     default: false,
   },
-  invalid: {
+  labelClass: ClassProp,
+  indeterminate: {
     type: Boolean,
     default: false,
   },
 });
 
-const emit = defineEmits<{
-  (event: 'update:modelValue', param: InputHTMLAttributes['checked']): void;
-}>();
+defineOptions({
+  inheritAttrs: false,
+});
+const model = defineModel<InputHTMLAttributes['checked']>({ default: false });
+const checkboxRef = useTemplateRef('checkboxRef');
 
-const { modelValue } = toRefs(props);
+const isChecked = ref();
+const isIndeterminate = ref();
 
-const proxyChecked = computed({
-  get: () => modelValue?.value,
-  set: (value) => emit('update:modelValue', value),
-});
+watch(
+  [model, () => indeterminate],
+  async () => {
+    await nextTick();
```

Comment:
nextTick is required since vue watch and what we get from ref is one tick behind, in other words when we have an `indeterminate=true` state, it will be `false`, but when we then set `indeterminate=false` we will get `true` in watcher (same happens in template), so pushing event to end of stack event in vue, fix this issue.

Reply:
have you tried creating [post watcher](https://vuejs.org/guide/essentials/watchers.html#post-watchers)?